src/runtime-rs/crates/hypervisor/src/firecracker/inner.rs

```diff
@@ -0,0 +1,217 @@
+        capabilities.set(CapabilityBits::BlockDeviceSupport);
+        FcInner {
+            id: String::default(),
+            asock_path: String::default(),
+            state: VmmState::NotReady,
+            config: Default::default(),
+            pid: None,
+            netns: None,
+            vm_path: String::default(),
+            client: Client::unix(),
+            jailer_root: String::default(),
+            jailed: false,
+            run_dir: String::default(),
+            pending_devices: vec![],
+            capabilities,
+        }
+    }
+
+    pub(crate) async fn prepare_vmm(&mut self, netns: Option<String>) -> Result<()> {
+        let mut cmd: Command;
+        self.netns = netns.clone();
+        match self.jailed {
+            true => {
+                debug!(sl!(), "[Firecracker] Running Jailed");
+                cmd = Command::new(&self.config.jailer_path);
+                let api_socket = ["/run/", FC_API_SOCKET_NAME].join("/");
+                let args = [
+                    "--id",
+                    &self.id,
+                    "--gid",
+                    "0",
+                    "--uid",
+                    "0",
+                    "--exec-file",
+                    &self.config.path,
+                    "--chroot-base-dir",
+                    &self.jailer_root,
+                    "--",
+                    "--api-sock",
+                    &api_socket,
+                ];
+                cmd.args(args);
+            }
+            false => {
+                debug!(sl!(), "[Firecracker] Running non-Jailed");
+                cmd = Command::new(&self.config.path);
+                cmd.args(["--api-sock", &self.asock_path]);
+            }
+        }
+        debug!(sl!(), "[Firecracker] Exec: {:?}", cmd);
+
+        // Make sure we're in the correct Network Namespace
+        unsafe {
+            let _pre = cmd.pre_exec(move || {
+                if let Some(netns_path) = &netns {
+                    debug!(sl!(), "set netns for vmm master {:?}", &netns_path);
+                    let netns_fd = std::fs::File::open(netns_path);
+                    let _ = setns(netns_fd?.as_raw_fd(), CloneFlags::CLONE_NEWNET)
+                        .context("set netns failed");
+                }
+                Ok(())
+            });
+        }
+
+        let mut child = cmd.spawn()?;
+
+        match child.id() {
+            Some(id) => {
+                let cur_tid = nix::unistd::gettid().as_raw() as u32;
+                info!(
+                    sl!(),
+                    "[Firecracker] VMM spawned successfully: PID: {:?}, current TID: {:?}",
+                    id,
+                    cur_tid
+                );
+                self.pid = Some(id);
+            }
+            None => {
+                let exit_status = child.wait().await?;
+                error!(
+                    sl!(),
+                    "[Firecracker] Process exited, status: {:?}", exit_status
```

Comment:
Maybe you can redefine a sl! macro including the Firecracker sub system instead of writing them in the log message every time?

Reply:
ack!